### Lab Report 5 - Link Analysis

In [1]:
# https://intellipaat.com/blog/data-science-apriori-algorithm/#:~:text=Demand%20Skills%20Now%20!-,What%20Is%20an%20Apriori%20Algorithm%3F,bread%7D%20must%20also%20be%20frequent.

In [2]:
pip install apyori

  Using cached apyori-1.1.2.tar.gz (8.6 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
  Created wheel for apyori: filename=apyori-1.1.2-py3-none-any.whl size=5975 sha256=8f26c6f44e3a1edc93d7309deda738ba3ed231da317c5440836398f7546f52dd
  Stored in directory: c:\users\user\appdata\local\pip\cache\wheels\77\3d\a6\d317a6fb32be58a602b1e8c6b5d6f31f79322da554cad2a5ea
Successfully built apyori
Note: you may need to restart the kernel to use updated packages.


In [3]:
# Step 1: Import the libraries

import numpy as np
import pandas as pd
from apyori import apriori

In [4]:
# Step 2: Load the dataset

store_data = pd.read_csv('Day1.csv', header=None)

In [5]:
#Step 3: Have a glance at the records

store_data

0      1      2       3     4      5
0   Wine  Chips  Bread  Butter  Milk  Apple
1   Wine    NaN  Bread  Butter  Milk    NaN
2    NaN    NaN  Bread  Butter  Milk    NaN
3    NaN  Chips    NaN     NaN   NaN  Apple
4   Wine  Chips  Bread  Butter  Milk  Apple
5   Wine  Chips    NaN     NaN  Milk    NaN
6   Wine  Chips  Bread  Butter   NaN  Apple
7   Wine  Chips    NaN     NaN  Milk    NaN
8   Wine    NaN  Bread     NaN   NaN  Apple
9   Wine    NaN  Bread  Butter  Milk    NaN
10   NaN  Chips  Bread  Butter   NaN  Apple
11  Wine    NaN    NaN  Butter  Milk  Apple
12  Wine  Chips  Bread  Butter  Milk    NaN
13  Wine    NaN  Bread     NaN  Milk  Apple
14  Wine    NaN  Bread  Butter  Milk  Apple
15  Wine  Chips  Bread  Butter  Milk  Apple
16   NaN  Chips  Bread  Butter  Milk  Apple
17   NaN  Chips    NaN  Butter  Milk  Apple
18  Wine  Chips  Bread  Butter  Milk  Apple
19  Wine    NaN  Bread  Butter  Milk  Apple
20  Wine  Chips  Bread     NaN  Milk  Apple
21   NaN  Chips    NaN     NaN   NaN    NaN

In [6]:
# Step 4: Look at the shape
store_data.shape

(22, 6)

In [7]:
# Step 5: Convert Pandas DataFrame into a list of lists

# Converting the pandas dataframe into a list of lists
records = []
for i in range (0,22):
    records.append([str(store_data.values[i,j]) for j in range (0, 6)])

In [29]:
# Step 6: Build the Apriori model

association_rules = apriori(records, min_support=0.5, min_confidence=0.7, min_lift=1.2, min_length=2)
association_results = list(association_rules)

Based on the code above, the parameters are meant by:
(a) records It will display the list of transactions
(b)min_support 0.5 is calculated by taking the 3 transactions in a day to the total number of
transactions
(c)min_confidence 0.7 is the mininum confidence. However, it can be changed as per the business
problem.
(d)min_lift 1.2 is the minimum lift
(e)min_length It will take 2 products for the association as the minimum number of products

In [30]:
#Step 7: Print out the number of rules

# Getting the number of rules
print(len(association_results))

1


In [31]:
#Step 8: Have a glance at the rule
print(association_results)

[RelationRecord(items=frozenset({'Bread', 'Butter', 'Milk'}), support=0.5, ordered_statistics=[OrderedStatistic(items_base=frozenset({'Butter'}), items_add=frozenset({'Bread', 'Milk'}), confidence=0.7333333333333334, lift=1.241025641025641), OrderedStatistic(items_base=frozenset({'Bread', 'Milk'}), items_add=frozenset({'Butter'}), confidence=0.8461538461538461, lift=1.241025641025641)])]


The support value for the first rule is 0.5. This number is obtained by dividing the number of
transactions containing “Bread”, “Butter” and “Milk” by the total number of transactions which
is 22.
The first confidence level for the rule is 0.733333, which shows that out of all the transactions that
contain “Butter”, 73.3333 % contain both “Bread”and “Milk” too. Besides, the second confidence
level for the rule is 0.846154, which shows that out of all the transactions that contain both “Bread”
and “Milk”, 84.6154 % contain “Butter” too.
The first lift of 1.241 tells us that both “Bread”, “Milk” is 1.241 times more likely to be bought by
the customers who buy “Butter” compared to the default likelihood sale of “Bread” and “Milk”.
3
Besides, the second lift of 1.241 tells us that “Butter” is 1.241 times more likely to be bought by the
customers who buy both “Bread” and “Milk” compared to the default likelihood sale of “Butter”.